src/relay/op/nn/nn.h

```diff
@@ -49,7 +49,7 @@ bool DenseRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
     // validate the weight shape is proper if defined
     // Assign weight type
     Array<IndexExpr> wshape({param->units, dshape[dshape.size() - 1]});
-    reporter->Assign(types[1], TensorTypeNode::make(wshape, data->dtype));
+    reporter->Assign(types[1], TensorTypeNode::make(wshape, weight->dtype));
```

Comment:
Please add a check if weight is defined. Otherwise, it can lead to seg fault (which might be the reason of failure in CI)